hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -927,6 +945,9 @@ def timedelta64_dtypes(
     )
 
 
+# TODO: we should uncap max_len here, and for unicode/void below.
+# Also allow generating undetermined-width dtypes like "S" / "S0"? Possibly with
+# a new parameter allow_undetermined?
```

Comment:
Weakly against uncapped dtypes; they make sense as an aid to interactive use (eg notebooks) but aren't actually a dtype that an array can have.

Reply:
On the flipside, they \*are* a valid dtype, and I think we should avoid compromising completeness unless there's a reason to. Generating them seems harmless if you're passing directly to `np.array(dtype=)` (equivalent to the largest elem size), and if you're testing a function which accepts generic `np.dtype` then imo that's the place where you most want hypothesis to generate weird dtypes like this.

Reply:
Yeah, I see your point, but I can't think of an API that I like more than writing `| st.just(np.dtype("S"))`!  Perhaps we can just document that trick, and explain why/when it's useful?